.github/workflows/changelog-by-pull.yml

```diff
@@ -0,0 +1,50 @@
+name: Changelog by PR
+on:
+  pull_request_target:
+    types:
+      - "closed"
+      - "edited"
+    branches:
+      - "main"
+jobs:
+  condition_check:
+    name: Check conditions
+    runs-on: ubuntu-latest
+    # Conditions:
+    #   - the PR is edited or merged
+    #   - it is not a changelog PR
+    #   - and its milestone is open
+    if: |
+      (
+        github.event.pull_request &&
+        !contains(github.event.pull_request.labels.*.name, 'changelog')
+      ) && (
+        github.event.pull_request.state == 'closed'         &&
+        github.event.pull_request.merged                    &&
+        github.event.pull_request.milestone.state == 'open'
+      )
```

Comment:
I don't get it, why this `if` needs a separate job and not in a `changelog` job?

Reply:
Agreed, separate condition job is redundant. Fixed.